kedro/io/core.py

```diff
@@ -595,6 +595,13 @@ def _get_versioned_path(self, version: str) -> PurePosixPath:
         return self._filepath / version / self._filepath.name
 
     def load(self) -> _DO:
+        try:
+            if hasattr(self, "_fs"):
+                self._fs.exists(str(self._filepath))
+        except PermissionError as err:
+            raise DataSetError(
+                f"Cannot load versioned dataset '{self._filepath}' due to insufficient permission."
+            ) from err
```

Comment:
Could you explain how this works? The original error `VersionNotFound` when it does not has sufficient permission. Would this error `PermissionError` be triggered?